### PR TITLE
Transaction support for repository and criterias

### PR DESCRIPTION
**Pull request by cosmin-ichim**
Adds transaction support to repositoryBase and criterias

### CODE REVIEW

File: RepositoryBase.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ class RepositoryBase extends PObject {
     /**
      * The model used by this Repository.
      *
-     * @property {puzzleframework.core.db.Model}
+     * @property {sequelize.Model}
```
Comment (vduricu) [inline RepositoryBase.js:31]: Why did you change it to sequelize.Model?

**Review by vduricu — CHANGES_REQUESTED** [review]

File: RepositoryBase.js (hunk 1 above)
Comment (cosmin-ichim) [reply RepositoryBase.js:31]: puzzle.models.get returns the sequelize model from the inside of puzzleframework Model:
```
// Model loader class
get(entity) { 
    return this._models[entity]
      ? this._models[entity].model
      : null;
  }
```
